### PR TITLE
WIP: Add code intelligence integration

### PR DESCRIPTION
**Pull request by lava**
###  :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: .code-intelligence/fuzz_targets/cli_parser.cpp.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "-msse3",
+    "-mssse3",
+    "-msse4.1",
+    "-msse4.2",
+    "-fno-omit-frame-pointer",
+    "-Wall",
+    "-fdebug-prefix-map=/vast-fuzz=.",
+    "-std=gnu++17",
+    "-MD",
+    "-Wl,-rpath,/vast-fuzz/build/lib: build/lib/libvast.so.2021.02 build/lib/libcaf_openssl.so.0.17.6 build/lib/libcaf_io.so.0.17.6 build/lib/libcaf_core.so.0.17.6 /usr/lib/x86_64-linux gnu/libssl.so /usr/lib/x86_64-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libflatbuffers.a /usr/lib/x86_64-linux-gnu/libspdlog.so.1.5.0 /usr/lib/x86_64-linux-gnu/libfmt.a /usr/lib/x86_64-linux-gnu/libsimdjson.so.4.0.0"
```
Comment (dominiklohmann) [inline .code-intelligence/fuzz_targets/cli_parser.cpp.json:35]: Can we invoke a custom CMake build target instead?

File: .github/workflows/code-intelligence.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          test_collection: "projects/initial-fuzzing-c6e8b32e/campaigns/parse_query-20e3604d"
+          git_reference: ${{ github.sha }}
+          fuzzing_server_address: ${{ env.FUZZING_SERVER_ADDRESS }}          
+      - id: monitor-fuzzing
+        name: Fuzzing
+        uses: CodeIntelligenceTesting/github-actions/monitor-fuzzing@v3
+        with:
+          ci_fuzz_api_token: ${{ secrets.CI_FUZZ_API_TOKEN }}
+          test_collection_run: ${{ steps.start-fuzzing.outputs.test_collection_run }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (dominiklohmann) [inline .github/workflows/code-intelligence.yaml:32]: Use `github.token` instead.

File: .github/workflows/code-intelligence.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          dashboard_address: ${{ env.WEB_APP_ADDRESS }}
+      - id: report-coverage
+        name: Report Coverage
+        uses: CodeIntelligenceTesting/github-actions/report-coverage@v3
+        if: ${{ github.event_name == 'pull_request' && (success() || failure()) }}
+        with:
+          ci_fuzz_api_token: ${{ secrets.CI_FUZZ_API_TOKEN }}
+          test_collection_run: ${{ steps.start-fuzzing.outputs.test_collection_run }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          pull_request_number: ${{ github.event.pull_request.number }}
```
Comment (dominiklohmann) [inline .github/workflows/code-intelligence.yaml:46]: The PR number will be empty for push triggers.

**Review by dominiklohmann — COMMENTED** [review]
Some questions on the setup.

File: .code-intelligence/campaigns/parse_query-20e3604d.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "parse_query-20e3604d",
```
Comment (lava) [inline .code-intelligence/campaigns/parse_query-20e3604d.json:2]: The numeric identifiers are autogenerated but I'm not sure if they're required, I'm tempted to just remove them and see what happens.

**Comment by lava** [comment]
We're now hitting the "git describe" error in the Code Intelligence build, which should be fixed by #1570

**Comment by mavam** [comment]
[ch24770]

**Comment by lava** [comment]
Closing this since we're planning to keep the fuzzing functionality in a separate `tenzir/vast-fuzzing` repo.

**Pull request closed without merging**